starlight-rabbitmq/src/main/java/com/datastax/oss/starlight/rabbitmq/GatewayService.java

```diff
@@ -196,7 +199,11 @@ public void start(boolean startChannels) throws Exception {
     pulsarAdmin = createAdminInstance();
     curator = createCuratorInstance();
     metadataModel = ModeledFramework.wrap(AsyncCuratorFramework.wrap(curator), METADATA_MODEL_SPEC);
-    executor.scheduleWithFixedDelay(this::loadContext, 0, 100, TimeUnit.MILLISECONDS);
+
+    persistentWatcher = new PersistentWatcher(curator, "/config", true);
+    persistentWatcher.start();
+    persistentWatcher.getListenable().addListener(event -> loadContext());
```

Comment:
what about logging the "event" ? at INFO level ?

then you can manually test what happens when you restart ZK (ConnectionLoss)

Reply:
I think that you have to register here:
https://github.com/apache/curator/blob/7f9e9cc48e04e4f6cdbea47b98ba2c602059d973/curator-recipes/src/main/java/org/apache/curator/framework/recipes/watch/PersistentWatcher.java#L143

to get notifications about "reconnections", otherwise when the client is reconnecting you are not notified of missed changes

Reply:
done. PTAL